File: rb/coin_changer.rb
class Changer
  attr_accessor :change
  QUARTER = 25
  DIME = 10
  NICKEL = 5
  PENNY = 1

  def initialize
    @change = []
  end

  def changer(num)
    until num == 0
      if num > 100
        puts "that's too high"
        exit
      elsif num / 25 >= 1
        @change.push(QUARTER)
        num = num - 25
      elsif num / 10 >= 1
        @change.push(DIME)
        num = num - 10
      elsif num / 5 >= 1
        @change.push(NICKEL)
        num = num - 5
      else
        @change.push(PENNY)
        num = num - 1
      end
    end
    puts @change
  end
end

Changer.new.changer(99)

File: js/coinChanger.js
/*global console */

'use strict';

var change = [];
var changer = function(num) {
  while (num > 0) {
    if (num > 100) {
      console.log('that\'s too high');
      return;
    }
    else if (num / 25 >= 1) {
      change.push(25);
      num =  num - 25;
    }
    else if (num / 10 >= 1) {
      change.push(10);
      num = num - 10;
    }
    else if (num / 5 >= 1) {
      change.push(5);
      num = num - 5;
    }
    else {
      change.push(1);
      num = num - 1;
    }
  }
  return console.log(change);
};

changer(99);
